### PR TITLE
[TRAFFIC-9336]: Add `confluent network ip-address list` command

### PR DESCRIPTION
**Pull request by syoung-cflt**
What
----

Note that this is merging into the feature branch https://github.com/confluentinc/cli/tree/traffic-8850-network-cli. We will merge commits into main from this feature branch after all of the commands for network are implemented

* Add `confluent network ip-address`
* Add `confluent network ip-address list`

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: test/test-server/networking_handlers.go [hunk 1]
```diff
@@ -1114,3 +1125,53 @@ func handleNetworkingPrivateLinkAttachmentConnectionList(t *testing.T) http.Hand
...
+func getIpAddress(ipPrefix, cloud, region string, services []string) networkingipv1.NetworkingV1IpAddress {
+	return networkingipv1.NetworkingV1IpAddress{
+		Kind:        networkingipv1.PtrString("IpAddress"),
+		AddressType: networkingipv1.PtrString("EGRESS"),
+		IpPrefix:    networkingipv1.PtrString(ipPrefix),
+		Cloud:       networkingipv1.PtrString(cloud),
+		Region:      networkingipv1.PtrString(region),
+		Services: &networkingipv1.NetworkingV1Services{
+			Items: services,
+		},
```
Comment (brianstrauch) [inline test/test-server/networking_handlers.go:1175]:
```suggestion
		Services: &networkingipv1.NetworkingV1Services{Items: services},
```

File: internal/network/command_ip_address_list.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/confluentinc/cli/v3/pkg/output"
+)
+
+type listIpAddressHumanOut struct {
+	AddressType string `human:"Address Type"`
+	IpPrefix    string `human:"IP Prefix"`
+	Cloud       string `human:"Cloud"`
+	Region      string `human:"Region"`
+	Services    string `human:"Services"`
+}
```
[the remark this answers is not in the document]
Comment (syoung-cflt) [reply internal/network/command_ip_address_list.go:18]: Updated!

File: test/test-server/networking_handlers.go (hunk 1 above)
Comment (syoung-cflt) [reply test/test-server/networking_handlers.go]: Fixed

**Review by brianstrauch — APPROVED** [review]

**Comment by ejagade** [comment]
@kabraham13 wanted to confirm if it is OK to have this `list` command without any filters, which means it always displays all egress ips

**Comment by kabraham13** [comment]
> @kabraham13 wanted to confirm if it is OK to have this `list` command without any filters, which means it always displays all egress ips

Is it not possible to add filters @ejagade ? I am worried that the results will be a very long list and hard to parse without it

**Comment by ejagade** [comment]
> > @kabraham13 wanted to confirm if it is OK to have this `list` command without any filters, which means it always displays all egress ips
> 
> Is it not possible to add filters @ejagade ? I am worried that the results will be a very long list and hard to parse without it

The bug in our OpenAPI generator logic is still open which prevents adding filter to any List command. For other resources it didn't matter much. But for ListIPs, it becomes a blocker

**Comment by kabraham13** [comment]
Got it @ejagade . Does APIF have any sort of line of sight to getting the bug fixed? Maybe we don't need to block our launch of the `list` command for this but it would be good to know what timeline it would go away since we'll need to go back and update this command

**Comment by syoung-cflt** [comment]
@kabraham13 I cc'd you in the slack thread regarding this bug. We are still clarifying with the APIF team on the timeline.

**Review by ejagade — APPROVED** [review]
